core/dyn-configs/src/lib.rs

```diff
@@ -0,0 +1,15 @@
+#![doc = include_str!("../README.md")]
+
+use std::sync::atomic::{AtomicU64, Ordering};
+
+// NOTE: AtomicU64 is the same unit as BlockHeight, and use to store the expected blockheight to
+// shutdown
+pub static EXPECTED_SHUTDOWN_AT: AtomicU64 = AtomicU64::new(0);
+
+pub fn reload(expected_shutdown: Option<u64>) {
+    if let Some(expected_shutdown) = expected_shutdown {
```

Comment:
maybe in separate PR -- I'd add a prometheus metric to display the current value of expected shutdown.
(feel free to just create an issue for that - if you don't have time to implement)